travis/travis_weblate.py

```diff
@@ -0,0 +1,248 @@
+        self._travis_build_dir = os.environ.get("TRAVIS_BUILD_DIR", "../..")
+        self._odoo_version = os.environ.get("VERSION")
+        self._odoo_branch = os.environ.get("ODOO_BRANCH")
+        self._langs = (parse_list(os.environ.get("LANG_ALLOWED")) if
+                       os.environ.get("LANG_ALLOWED", False) else [])
+        self._odoo_full = os.environ.get("ODOO_REPO", "odoo/odoo")
+        self._server_path = get_server_path(self._odoo_full,
+                                            (self._odoo_branch or
+                                             self._odoo_version),
+                                            self._travis_home)
+        self._addons_path = get_addons_path(self._travis_home,
+                                            self._travis_build_dir,
+                                            self._server_path)
+        self._connection_context = context_mapping.get(
+            self._odoo_version, Odoo10Context)
+        self._apply_patch_odoo()
+        self._get_modules_installed()
+
+    def _check(self):
+        self.wl_api._check()
+        self.gh_api._check()
+
+    def _apply_patch_odoo(self):
+        """This patch is necessary because the weblate does not check which
+        word and the translated are the same to use it in its percentage of
+        translated"""
+        paths = [os.path.join('openerp', 'tools', 'translate.py'),
+                 os.path.join('odoo', 'tools', 'translate.py')]
+        for path in paths:
+            s_file = os.path.join(self._server_path, path)
+            if not os.path.isfile(s_file):
+                continue
+            cmd = ["sed", "-i", "-e",
+                   r"s/translation'] = src/translation'] = ''/g",
+                   s_file]
+            print " ".join(cmd)
+            subprocess.call(cmd)
+
+    def _get_modules_installed(self):
+        self._installed_modules = []
+        modules_found = []
+        for name in ('__openerp__.py', '__manifest__.py'):
+            modules = glob.glob('%s/**/%s' % (self._travis_build_dir, name))
+            if not modules:
+                continue
+            modules_found.extend([
+                os.path.dirname(module).split('/')[-1] for module in
+                modules])
+        with self._connection_context(self._server_path, self._addons_path,
+                                      "openerp_test") as odoo_context:
+            odoo_context.cr.execute("select name from ir_module_module"
+                                    " where state = 'installed' and "
+                                    "name in %s", (tuple(modules_found),))
+            modules = odoo_context.cr.dictfetchall()
+            self._installed_modules = [module['name'] for module in modules]
+
+    def _generate_odoo_po_files(self, component):
+        generated = False
+        with self._connection_context(self._server_path, self._addons_path,
+                                      "openerp_test") as odoo_context:
```

Comment:
Maybe we can use Odoo's config value like here: https://github.com/OCA/bank-payment/blob/7a5301f8593f065fb4048d3adca22fd71c7ab212/account_payment_transfer_reconcile_batch/models/payment_order.py#L28